bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABridgeHandler.java

```diff
@@ -0,0 +1,261 @@
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlugwiseHABridgeHandler} class is responsible for handling
+ * commands and status updates for the Plugwise Home Automation bridge.
+ * Extends @{link BaseBridgeHandler}
+ *
+ * @author Bas van Wetten - Initial contribution
+ * @author Leo Siepel - finish initial contribution
+ * 
+ */
+
+@NonNullByDefault
+public class PlugwiseHABridgeHandler extends BaseBridgeHandler {
+
+    // Private Static error messages
+
+    private static final String STATUS_DESCRIPTION_COMMUNICATION_ERROR = "Error communicating with the Plugwise Home Automation controller";
+    private static final String STATUS_DESCRIPTION_TIMEOUT = "Communication timeout while communicating with the Plugwise Home Automation controller";
+    private static final String STATUS_DESCRIPTION_CONFIGURATION_ERROR = "Invalid or missing configuration";
+    private static final String STATUS_DESCRIPTION_INVALID_CREDENTIALS = "Invalid username and/or password - please double-check your configuration";
+    private static final String STATUS_DESCRIPTION_INVALID_HOSTNAME = "Invalid hostname - please double-check your configuration";
+
+    // Private member variables/constants
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable volatile PlugwiseHAController controller;
+
+    private final HttpClient httpClient;
+    private final Logger logger = LoggerFactory.getLogger(PlugwiseHABridgeHandler.class);
+
+    // Constructor
+
+    public PlugwiseHABridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    // Public methods
+
+    @Override
+    public void initialize() {
+        PlugwiseHABridgeThingConfig bridgeConfig = getConfigAs(PlugwiseHABridgeThingConfig.class);
+
+        if (this.checkConfig(bridgeConfig)) {
+            logger.debug("Initializing the Plugwise Home Automation bridge handler with config = {}", bridgeConfig);
+            try {
+                this.controller = new PlugwiseHAController(httpClient, bridgeConfig.getHost(), bridgeConfig.getPort(),
+                        bridgeConfig.getUsername(), bridgeConfig.getsmileId());
+                scheduleRefreshJob(bridgeConfig);
+            } catch (PlugwiseHAException e) {
+                logger.debug("Unknown error while configuring the Plugwise Home Automation Controller", e);
+                updateStatus(OFFLINE, CONFIGURATION_ERROR, e.getMessage());
+            } catch (RuntimeException e) {
+                logger.debug("Unknown error while configuring the Plugwise Home Automation Controller", e);
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, e.getMessage());
+            }
```

Comment:
Unchecked exceptions don't need to be cought here. The framework will do it and set the Thing status accordingly.

Reply:
fixed